src/coinbase/webhook.ts

```diff
@@ -0,0 +1,180 @@
+import { Webhook as WebhookModel, WebhookEventType, WebhookEventFilter } from "../client/api";
+import { Coinbase } from "./coinbase";
+import { InternalError } from "./errors";
+
+/**
+ * A representation of a Webhook,
+ * which provides methods to create, list, update, and delete webhooks that are used to receive notifications of specific events.
+ */
+export class Webhook {
+  private model: WebhookModel | null;
+
+  /**
+   * Initializes a new Webhook object.
+   *
+   * @param model - The underlying Webhook object.
+   * @throws {InternalError} If the model is not provided.
+   */
+  private constructor(model: WebhookModel) {
+    if (!model) {
+      throw new InternalError("Webhook model cannot be empty");
+    }
+    this.model = model;
+  }
+
+  /**
+   * Returns a new Webhook object. Do not use this method directly. Instead, Webhook.create(...)
+   *
+   * @constructs Webhook
+   * @param model - The underlying Webhook model object
+   * @returns A Webhook object.
+   */
+  public static init(model: WebhookModel): Webhook {
+    return new Webhook(model);
+  }
+
+  /**
+   * Creates a new webhook for a specified network.
+   *
+   * @param networkId - The network ID for which the webhook is created.
+   * @param notificationUri - The URI where notifications should be sent.
+   * @param eventType - The type of event for the webhook.
+   * @param eventFilters - Filters applied to the events that determine which specific events trigger the webhook.
+   * @returns A promise that resolves to a new instance of Webhook.
+   */
+  public static async create(
+    networkId,
+    notificationUri,
+    eventType,
+    eventFilters,
```

Comment:
can we add types for these params. Thanks!

Reply:
done